filecoin-proofs/src/api/seal.rs

```diff
@@ -444,16 +445,35 @@ pub fn seal_commit_phase1_inner<T: AsRef<Path>, Tree: 'static + MerkleTreeTrait>
         deserialize(&p_aux_bytes)
     }?;
 
+    let vanilla_params = setup_params(porep_config)?;
+
     let t_aux = {
         let t_aux_path = cache_path.as_ref().join(CacheKey::TAux.to_string());
-        let t_aux_bytes = fs::read(&t_aux_path)
-            .with_context(|| format!("could not read file t_aux={:?}", t_aux_path))?;
-
-        let mut res: TemporaryAux<_, _> = deserialize(&t_aux_bytes)?;
-
-        // Switch t_aux to the passed in cache_path
-        res.set_cache_path(cache_path);
-        res
+        match fs::read(&t_aux_path) {
```

Comment:
As-is, this won't work for Sector Updates (and maybe other parts?).  I don't think that path will be much more complicated, but does require more testing, as `t_aux` is very much not optional in at least that part of the code.

Reply:
Isn't the Sector Updates part self-contained at https://github.com/filecoin-project/rust-fil-proofs/blob/2ec0a0fe4005a6f2f6659bbbc167a1d41ee855e0/filecoin-proofs/src/api/update.rs? This part should only concern the C1.

So would be adding a test that removes the `t_aux` and then runs C1 be sufficient?

Reply:
The sector updates code currently relies on t_aux being available and if it's not available from an earlier step in sealing, future sector updates won't work.

Reply:
Sorry, now I get it. The problem is **not** about the code paths. Each works fine independently. The problem is that if someone seals a CC sector and removes their `t_aux`, they could then still C1, but they later could **not** do a sector update.

This means that I also need to take care of the sector update code path. Would adding a similar fallback there be an OK solution?